test/MathUtils.test.cpp

```diff
@@ -0,0 +1,237 @@
+
+	//Right
+	EXPECT_EQ(false, NAS2D::isPointInRect(6, 4, 5, 5, 1, 1));
+	EXPECT_EQ(true, NAS2D::isPointInRect(6, 5, 5, 5, 1, 1));
+	EXPECT_EQ(true, NAS2D::isPointInRect(6, 6, 5, 5, 1, 1));
+	EXPECT_EQ(false, NAS2D::isPointInRect(6, 7, 5, 5, 1, 1));
+
+	//Center
+	EXPECT_EQ(false, NAS2D::isPointInRect(4, 6, 5, 5, 2, 2));
+	EXPECT_EQ(true, NAS2D::isPointInRect(5, 6, 5, 5, 2, 2));
+	EXPECT_EQ(true, NAS2D::isPointInRect(6, 6, 5, 5, 2, 2));
+	EXPECT_EQ(true, NAS2D::isPointInRect(7, 6, 5, 5, 2, 2));
+	EXPECT_EQ(false, NAS2D::isPointInRect(8, 6, 5, 5, 2, 2));
+}
+
+TEST(MathUtils, isPointInRectClass)
+{
+	using namespace NAS2D;
+	//Neghbors
+	EXPECT_EQ(false, NAS2D::isPointInRect({4, 4}, {5, 5, 1, 1}));
+	EXPECT_EQ(false, NAS2D::isPointInRect({5, 4}, {5, 5, 1, 1}));
+	EXPECT_EQ(false, NAS2D::isPointInRect({6, 4}, {5, 5, 1, 1}));
+	EXPECT_EQ(false, NAS2D::isPointInRect({7, 4}, {5, 5, 1, 1}));
+	EXPECT_EQ(false, NAS2D::isPointInRect({7, 5}, {5, 5, 1, 1}));
+	EXPECT_EQ(false, NAS2D::isPointInRect({7, 6}, {5, 5, 1, 1}));
+	EXPECT_EQ(false, NAS2D::isPointInRect({7, 7}, {5, 5, 1, 1}));
+	EXPECT_EQ(false, NAS2D::isPointInRect({6, 7}, {5, 5, 1, 1}));
+	EXPECT_EQ(false, NAS2D::isPointInRect({5, 7}, {5, 5, 1, 1}));
+	EXPECT_EQ(false, NAS2D::isPointInRect({4, 7}, {5, 5, 1, 1}));
+	EXPECT_EQ(false, NAS2D::isPointInRect({4, 6}, {5, 5, 1, 1}));
+	EXPECT_EQ(false, NAS2D::isPointInRect({4, 5}, {5, 5, 1, 1}));
+	EXPECT_EQ(false, NAS2D::isPointInRect({4, 4}, {5, 5, 1, 1}));
+
+	//Top
+	EXPECT_EQ(false, NAS2D::isPointInRect({4, 5}, {5, 5, 1, 1}));
+	EXPECT_EQ(true, NAS2D::isPointInRect({5, 5}, {5, 5, 1, 1}));
+	EXPECT_EQ(true, NAS2D::isPointInRect({6, 5}, {5, 5, 1, 1}));
+	EXPECT_EQ(false, NAS2D::isPointInRect({7, 5}, {5, 5, 1, 1}));
+
+	//Bottom
+	EXPECT_EQ(false, NAS2D::isPointInRect({4, 6}, {5, 5, 1, 1}));
+	EXPECT_EQ(true, NAS2D::isPointInRect({5, 6}, {5, 5, 1, 1}));
+	EXPECT_EQ(true, NAS2D::isPointInRect({6, 6}, {5, 5, 1, 1}));
+	EXPECT_EQ(false, NAS2D::isPointInRect({7, 6}, {5, 5, 1, 1}));
+
+	//Left
+	EXPECT_EQ(false, NAS2D::isPointInRect({5, 4}, {5, 5, 1, 1}));
+	EXPECT_EQ(true, NAS2D::isPointInRect({5, 5}, {5, 5, 1, 1}));
+	EXPECT_EQ(true, NAS2D::isPointInRect({5, 6}, {5, 5, 1, 1}));
+	EXPECT_EQ(false, NAS2D::isPointInRect({5, 7}, {5, 5, 1, 1}));
+
+	//Right
+	EXPECT_EQ(false, NAS2D::isPointInRect({6, 4}, {5, 5, 1, 1}));
+	EXPECT_EQ(true, NAS2D::isPointInRect({6, 5}, {5, 5, 1, 1}));
+	EXPECT_EQ(true, NAS2D::isPointInRect({6, 6}, {5, 5, 1, 1}));
+	EXPECT_EQ(false, NAS2D::isPointInRect({6, 7}, {5, 5, 1, 1}));
+
+	//Center
+	EXPECT_EQ(false, NAS2D::isPointInRect({4, 6}, {5, 5, 2, 2}));
+	EXPECT_EQ(true, NAS2D::isPointInRect({5, 6}, {5, 5, 2, 2}));
+	EXPECT_EQ(true, NAS2D::isPointInRect({6, 6}, {5, 5, 2, 2}));
+	EXPECT_EQ(true, NAS2D::isPointInRect({7, 6}, {5, 5, 2, 2}));
+	EXPECT_EQ(false, NAS2D::isPointInRect({8, 6}, {5, 5, 2, 2}));
+}
+
+TEST(MathUtils, isRectInRectComponent)
+{
+	//Neghbors
+	EXPECT_EQ(false, NAS2D::isRectInRect(3, 3, 1, 1, 5, 5, 1, 1));
```

Comment:
Again, this test shows the unpacked parameter value variant of the method is much harder to read at the point of call. I'm thinking this variant should be dropped.